bindings/tflite/java/org/tensorflow/lite/Interpreter.java

```diff
@@ -0,0 +1,436 @@
+   * execution if any input tensors have been resized. This call is most useful in determining the
+   * shapes for any output tensors before executing the graph, e.g.,
+   *
+   * <pre>{@code
+   * interpreter.resizeInput(0, new int[]{1, 4, 4, 3}));
+   * interpreter.allocateTensors();
+   * FloatBuffer input = FloatBuffer.allocate(interpreter.getInputTensor(0),numElements());
+   * // Populate inputs...
+   * FloatBuffer output = FloatBuffer.allocate(interpreter.getOutputTensor(0).numElements());
+   * interpreter.run(input, output)
+   * // Process outputs...
+   * }</pre>
+   *
+   * @throws IllegalStateException if the graph's tensors could not be successfully allocated.
+   */
+  public void allocateTensors() {
+    if (nativeAllocateTensors() != 0) {
+      throw new IllegalStateException("Failed to allocate Tensors.");
+    }
+    tensorsAllocated = true;
+  }
+
+  /**
+   * Resizes idx-th input of the native model to the given dims.
+   *
+   * @throws IllegalArgumentException if {@code idx} is negative or is not smaller than the number
+   *     of model inputs; or if error occurs when resizing the idx-th input.
+   */
+  public void resizeInput(int inputIndex, @NonNull int[] dims) {
+    if (nativeResizeInputTensor(inputIndex, dims) != 0) {
+      throw new IllegalArgumentException("Unable to resize to input tensor.");
+    }
+    tensorsAllocated = false;
+  }
+
+  /** Gets the number of input tensors. */
+  public int getInputTensorCount() {
+    return nativeInputTensorCount();
+  }
+
+  /**
+   * Gets index of an input given the op name of the input.
+   *
+   * @throws IllegalArgumentException if {@code opName} does not match any input in the model used
+   *     to initialize the {@link Interpreter}.
+   */
+  public int getInputIndex(String opName) {
+    for (int i = 0; i < getInputTensorCount(); ++i) {
+      if (getInputTensor(i).name().equals(opName)) {
+        return i;
+      }
+    }
+    return -1;
+  }
+
+  /**
+   * Gets the Tensor associated with the provdied input index.
+   *
+   * @throws IllegalArgumentException if {@code inputIndex} is negtive or is not smaller than the
+   *     number of model inputs.
+   */
+  public Tensor getInputTensor(int index) {
+    if (index < 0 || index >= inputTensors.length) {
+      throw new IllegalArgumentException(String.format("Invalid input Tensor index: %d", index));
+    }
+    Tensor inputTensor = inputTensors[index];
+    if (inputTensor == null) {
+      inputTensor = inputTensors[index] = Tensor.inputFromIndex(nativeAddress, index);
+    }
+    return inputTensor;
+  }
+
+  /** Gets the number of output Tensors. */
+  public int getOutputTensorCount() {
+    return nativeOutputTensorCount();
+  }
+
+  /** Gets index of an output given the op name of the output or -1 if not found. */
+  public int getOutputIndex(String opName) {
+    for (int i = 0; i < getOutputTensorCount(); ++i) {
```

Comment:
If the value of `getOutputTensorCount()` does not change, you should either cache it in the class itself, or at least store it in a local variable here.

This public method shouldn't need to keep crossing JNI to get a constant integer value.